### ISSUE TITLE
demo2D00 not working

### ISSUE THREAD

**Issue by icaoberg** [icaoberg/docker-cellorganizer#17]
demo2D00 is not working properly

```
➜  docker-cellorganizer git:(master) docker exec $ID ./cellorganizer/demos/2D/demo2D00/demo2D00.sh
Warning: Name is nonexistent or not a directory: /home/murphylab/.mcrCache9.2/slml2i0/utilities/3D/LDDMM_fast/imrender/vgg
> In path (line 109)
  In addpath (line 88)
  In startup (line 3)
Warning: Name is nonexistent or not a directory: /home/murphylab/.mcrCache9.2/slml2i0/utilities/3D/LDDMM_fast/imrender/ojw
> In path (line 109)
  In addpath (line 88)
  In startup (line 4)
Running deployed version of slml2img
Attempting to read input file /home/murphylab/input.txt
Evaluating lines from input file
filenames={'/home/murphylab/cellorganizer/models/2D/nucleolus.mat', '/home/murphylab/cellorganizer/models/2D/lysosome.mat', '/home/murphylab/cellorganizer/models/2D/endosome.mat', '/home/murphylab/cellorganizer/models/2D/mitochondrion.mat' };

options.targetDirectory = pwd;

options.prefix = 'img';

options.compression = 'lzw';

options.debug = false;

options.verbose = true;

options.display = false;

options.numberOfSynthesizedImages = 1;



Setting synthesis option
Checking existence of log folder
Checking the validity of input files
Parsing SLML instancesChecking existence of temporary folder
Creating temporary folder in /home/murphylab/temp
Setting target directory to /home/murphylab
Setting prefix to: img
Setting number of synthesized images to 1
Setting compression to lzw
Setting protein model resolution
Protein model object(s) resolution set to
Synthesizing image 1. Please wait, this might take several minutes.
Setting model dimensionality to 2D
Checking all models have the same dimensionality
Generating cell shape
Removing temporary folder
Removing temporary files
Closing log file.
Closing log file
Finished synthesis
img/cell1/cell1.tiff: cannot open `img/cell1/cell1.tiff' (No such file or directory)
```
it does create the image

```
➜  docker-cellorganizer git:(master) docker exec -it f2b410f1b37a /bin/bash
murphylab@f2b410f1b37a:~$ tree ./img
./img
`-- cell1
    `-- cell1.tif

1 directory, 1 file
```

so the test is broken?

**Comment by icaoberg** [comment]
@bstkj I don't know if this is a bug. However I "solved" it by login in as root. However, I don't know if this is the right thing to do

```
➜  docker-cellorganizer git:(master) docker exec --user root $ID sh -c "cd ./cellorganizer/demos/2D/demo2D00/ && bash demo2D00.sh"
```

which outputs

```
Setting synthesis option
Checking existence of log folder
Checking the validity of input files
Parsing SLML instancesChecking existence of temporary folder
Creating temporary folder in /home/murphylab/cellorganizer/demos/2D/demo2D00/temp
Setting target directory to /home/murphylab/cellorganizer/demos/2D/demo2D00
Setting prefix to: img
Setting number of synthesized images to 1
Setting compression to lzw
Setting protein model resolution
Protein model object(s) resolution set to
Synthesizing image 1. Please wait, this might take several minutes.
Setting model dimensionality to 2D
Checking all models have the same dimensionality
Generating cell shape
Removing temporary folder
Removing temporary files
Closing log file.
Closing log file
Finished synthesis
/home/murphylab/cellorganizer/demos/2D/demo2D00/img/cell1/cell1.tif: TIFF image data, little-endian, direntries=14, height=777, bps=8, compression=LZW, PhotometricIntepretation=BlackIsZero, orientation=upper-left, width=757
```

**Comment by icaoberg** [comment]
I cannot run as `murphylab` due to a permission issue

```
root@f2b410f1b37a:/home/murphylab# chown -Rv murphylab:x cellorganizer/
chown: invalid group: 'murphylab:x'
root@f2b410f1b37a:/home/murphylab# chown -Rv murphylab:100 cellorganizer
ownership of 'cellorganizer/models/3D/mit.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D/lamp2.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D/tfr.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D/centro.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D/tub.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D/ellipsoid_model.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D/diffeomorphic/helamodel_9_16_15.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D/diffeomorphic/hela_nuc_10_15_15.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D/diffeomorphic/cell_model_nuc_align.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D/diffeomorphic/3t3_model.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D/diffeomorphic/hela_cell_10_15_15.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D/diffeomorphic' retained as murphylab:users
ownership of 'cellorganizer/models/3D/nuc.mat' retained as murphylab:users
ownership of 'cellorganizer/models/3D' retained as murphylab:users
ownership of 'cellorganizer/models/2D/lysosome.mat' retained as murphylab:users
ownership of 'cellorganizer/models/2D/nucleolus.mat' retained as murphylab:users
ownership of 'cellorganizer/models/2D/endosome.mat' retained as murphylab:users
ownership of 'cellorganizer/models/2D/mitochondrion.mat' retained as murphylab:users
ownership of 'cellorganizer/models/2D' retained as murphylab:users
ownership of 'cellorganizer/models' retained as murphylab:users
changed ownership of 'cellorganizer/demos/3D/demo3D02/demo3D02.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D02' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D01/demo3D01.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D01' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D06/demo3D06.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D06' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D19/demo3D19.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D19' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D00/demo3D00.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D00' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D47/demo3D47.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D47' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D09/demo3D09.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D09' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D20/demo3D20.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D20' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D10/demo3D10.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D10' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D21/demo3D21.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D21' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D11/demo3D11.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D11' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D08/demo3D08.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D08' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D07/demo3D07.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D07' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D12/demo3D12.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D12' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D03/demo3D03.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D03' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D04/demo3D04.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D/demo3D04' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/3D' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D00/demo2D00.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D00/log/201832722958.5117.log' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D00/log/201832722832.3018.log' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D00/log/201832723146.5211.log' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D00/log' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D00/img/cell1/cell1.tif' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D00/img/cell1' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D00/img' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D00/input.txt' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D00' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D01/demo2D01.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D01' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D04/demo2D04.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D04' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D05/demo2D05.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D05' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D02/demo2D02.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D02' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D03/demo2D03.sh' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D/demo2D03' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos/2D' from root:root to murphylab:100
changed ownership of 'cellorganizer/demos' from root:root to murphylab:100
ownership of 'cellorganizer' retained as murphylab:users
```

**Comment by icaoberg** [comment]
Changing permissions seems to do the trick

```
➜  docker-cellorganizer git:(master) docker exec --user murphylab $ID sh -c "cd ./cellorganizer/demos/2D/demo2D00/ && bash demo2D00.sh"
Warning: Name is nonexistent or not a directory: /home/murphylab/.mcrCache9.2/slml2i0/utilities/3D/LDDMM_fast/imrender/vgg
> In path (line 109)
  In addpath (line 88)
  In startup (line 3)
Warning: Name is nonexistent or not a directory: /home/murphylab/.mcrCache9.2/slml2i0/utilities/3D/LDDMM_fast/imrender/ojw
> In path (line 109)
  In addpath (line 88)
  In startup (line 4)
Running deployed version of slml2img
Attempting to read input file /home/murphylab/cellorganizer/demos/2D/demo2D00/input.txt
Evaluating lines from input file
filenames={'/home/murphylab/cellorganizer/models/2D/nucleolus.mat', '/home/murphylab/cellorganizer/models/2D/lysosome.mat', '/home/murphylab/cellorganizer/models/2D/endosome.mat', '/home/murphylab/cellorganizer/models/2D/mitochondrion.mat' };

options.targetDirectory = pwd;

options.prefix = 'img';

options.compression = 'lzw';

options.debug = false;

options.verbose = true;

options.display = false;

options.numberOfSynthesizedImages = 1;



Setting synthesis option
Checking existence of log folder
Checking the validity of input files
Parsing SLML instancesChecking existence of temporary folder
Creating temporary folder in /home/murphylab/cellorganizer/demos/2D/demo2D00/temp
Setting target directory to /home/murphylab/cellorganizer/demos/2D/demo2D00
Setting prefix to: img
Setting number of synthesized images to 1
Setting compression to lzw
Setting protein model resolution
Protein model object(s) resolution set to
Synthesizing image 1. Please wait, this might take several minutes.
Setting model dimensionality to 2D
Checking all models have the same dimensionality
Generating cell shape
Removing temporary folder
Removing temporary files
Closing log file.
Closing log file
Finished synthesis
/home/murphylab/cellorganizer/demos/2D/demo2D00/img/cell1/cell1.tif: TIFF image data, little-endian, direntries=14, height=777, bps=8, compression=LZW, PhotometricIntepretation=BlackIsZero, orientation=upper-left, width=757
```

**Comment by icaoberg** [comment]
Fixed by PR #18
